src/utils/work/getExtraDependencies.js

```diff
@@ -17,12 +17,17 @@ const getClusteringSettings = async (experimentId, body, dispatch, getState) =>
   return clusteringSettings;
 };
 
-// Check that the cell sets within the selected cellSetKey didn't change
+// Check that the cell sets within the selected selectedCellSet didn't change
 // e.g., if cell set was deleted we can't use cache
 const getSelectedCellSet = async (experimentId, body, dispatch, getState) => {
```

Comment:
I know you didn't implement it, but why is this function here? It feels like this is something specific to the particular actions and it does not belong to the more generic `getExtraDependencies` file. Conceptually, to me this looks similar to what you do with the new function called "findLoadedGenes",  which is called in the action file `loadDownsampledGeneExpression` so I am surprised that it is here.

Reply:
I am rpetty sure I added it, just not in this PR :P.

If you check how it is being used, you can see it is only used inside this file. It is a way to declare dependencies in work requests somewhat declaratively and in a single place. They are not really params but they still should affect the ETag, that's why they are here.

You can check how they are used by looking at `const dependencyGetters =`

Reply:
Fair enough. Maybe I just have a general dislike for the fact that we need to have something like `getExtraDependencies` -- I would expect that the UI wouldn't need to worry about what exactly is needed for a certain work request to be completed successfully. It is something outside of the scope of this PR, it will be good to discuss and see how to address though.

Reply:
I believe pol is working on something that will replace `getExtraDependencies` by adding them somehow into the body of the request

Reply:
Thanks for the explanation, it is clearer now. Yes, from what I know Pol is working on moving extraDependencies and the etag calculation to the API, which should help